compiler/rustc_type_ir/src/walk.rs

```diff
@@ -1,20 +1,21 @@
 //! An iterator over the type substructure.
 //! WARNING: this does not keep track of the region depth.
 
-use rustc_data_structures::sso::SsoHashSet;
 use smallvec::{SmallVec, smallvec};
 use tracing::debug;
 
-use crate::ty::{self, GenericArg, GenericArgKind, Ty};
+use crate::data_structures::SsoHashSet;
+use crate::inherent::*;
+use crate::{self as ty, GenericArgKind, Interner};
```

Comment:
nit: same as above, could you just `ty::GenericArgKind` at the usage sites rather than import it?